apps/iir_blur/filter.cpp

```diff
@@ -13,12 +14,54 @@
 
 using namespace Halide::Tools;
 
+namespace {
+
+enum DeviceState {
+    USING_METAL_OR_OPENCL,
+    NOT_METAL_OR_OPENCL,
+    ENV_VARIABLE_ABSENT,
+};
+DeviceState ensure_cuda_device() {
+    const auto hl_target = std::getenv("HL_TARGET");
+    if (hl_target == nullptr) {
+        printf("Warning: Environment variable HL_TARGET not specified. "
+               "Proceeding to the tests...\n");
+        return ENV_VARIABLE_ABSENT;
+    }
```

Comment:
You can get this same information like so:

```suggestion
    const char* hl_target = iir_blur_metadata()->target;
```

Reply:
Ah yes, this solution is far more elegant. Thanks!